source/NVDAObjects/UIA/__init__.py

```diff
@@ -1231,12 +1231,28 @@ def _get_states(self):
 			states.add(controlTypes.STATE_INVALID_ENTRY)
 		if self._getUIACacheablePropertyValue(UIAHandler.UIA_IsRequiredForFormPropertyId):
 			states.add(controlTypes.STATE_REQUIRED)
+
+		# Read-only state

```

Comment:
Could you please extract this whole section into a new private method `_getIsReadOnlyState()`